From 3a23484c6744b80070ea2bf5044ae75a517cbff3 Mon Sep 17 00:00:00 2001
From: Pierre-Jean TEXIER <texier.pj2@gmail.com>
Date: Tue, 20 Mar 2018 22:56:09 +0100
Subject: [PATCH 4/5] warp7: Add support to UpdateHub

Signed-off-by: Pierre-Jean TEXIER <texier.pj2@gmail.com>

---
 include/configs/warp7.h | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/include/configs/warp7.h b/include/configs/warp7.h
index 2a7789baa2..a6bb61b50b 100644
--- a/include/configs/warp7.h
+++ b/include/configs/warp7.h
@@ -96,6 +96,42 @@
 		   "fi; " \
 	   "fi"
 
+#undef CONFIG_BOOTCOMMAND
+
+/*
+ * UpdateHub configuration
+ */
+
+#define CONFIG_BOOTCOUNT_ENV
+
+/* Environment */
+#define UPDATEHUB_LOAD_OS_A     "load mmc 0:1 ${loadaddr} /boot/${image}; " \
+                                "load mmc 0:1 ${fdt_addr} /boot/${fdt_file}; "
+#define UPDATEHUB_FIND_ROOT_A   "part uuid mmc 0:1 uuid"
+
+#define UPDATEHUB_LOAD_OS_B     "load mmc 0:2 ${loadaddr} /boot/${image}; " \
+                                "load mmc 0:2 ${fdt_addr} /boot/${fdt_file}; "
+#define UPDATEHUB_FIND_ROOT_B   "part uuid mmc 0:2 uuid"
+
+#define UPDATEHUB_BOOTARGS      "root=PARTUUID=${uuid} rootfstype=ext4 " \
+                                "rootwait rw console=${console},${baudrate} "
+#define UPDATEHUB_BOOTCMD		"bootz ${loadaddr} - ${fdt_addr}"
+
+#include <configs/updatehub-common.h>
+
+#undef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	CONFIG_DFU_ENV_SETTINGS \
+	"image=zImage\0" \
+	"console=ttymxc0\0" \
+	"ethact=usb_ether\0" \
+	"fdt_high=0xffffffff\0" \
+	"initrd_high=0xffffffff\0" \
+	"fdt_file=imx7s-warp.dtb\0" \
+	"fdt_addr=0x83000000\0" \
+	"ip_dyn=yes\0" \
+	UPDATEHUB_ENV
+
 #define CONFIG_SYS_MEMTEST_START	0x80000000
 #define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + 0x20000000)
 
