From 77685a49915936d328c747d29d3885ccf626f756 Mon Sep 17 00:00:00 2001
From: Pierre-Jean TEXIER <texier.pj2@gmail.com>
Date: Tue, 3 Apr 2018 18:15:36 +0200
Subject: [PATCH 5/5] pico-imx7d: Add support to UpdateHub

Signed-off-by: Pierre-Jean TEXIER <texier.pj2@gmail.com>
---
 include/configs/pico-imx7d.h | 35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/include/configs/pico-imx7d.h b/include/configs/pico-imx7d.h
index 462c5bf..38800b0 100644
--- a/include/configs/pico-imx7d.h
+++ b/include/configs/pico-imx7d.h
@@ -84,6 +84,41 @@
 		"else run netboot; " \
 		"fi; " \
 	"else run netboot; fi"
+	
+#undef CONFIG_BOOTCOMMAND
+
+/*
+ * UpdateHub configuration
+ */
+
+#define CONFIG_BOOTCOUNT_ENV
+
+/* Environment */
+#define UPDATEHUB_LOAD_OS_A     "load mmc 0:1 ${loadaddr} /boot/${image}; " \
+                                "load mmc 0:1 ${fdt_addr} /boot/${fdt_file}; "
+#define UPDATEHUB_FIND_ROOT_A   "part uuid mmc 0:1 uuid"
+
+#define UPDATEHUB_LOAD_OS_B     "load mmc 0:2 ${loadaddr} /boot/${image}; " \
+                                "load mmc 0:2 ${fdt_addr} /boot/${fdt_file}; "
+#define UPDATEHUB_FIND_ROOT_B   "part uuid mmc 0:2 uuid"
+
+#define UPDATEHUB_BOOTARGS      "root=PARTUUID=${uuid} rootfstype=ext4 " \
+                                "rootwait rw console=${console},${baudrate} "
+#define UPDATEHUB_BOOTCMD		"bootz ${loadaddr} - ${fdt_addr}"
+
+#include <configs/updatehub-common.h>
+
+#undef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"image=zImage\0" \
+	"console=ttymxc4\0" \
+	"ethact=usb_ether\0" \
+	"fdt_high=0xffffffff\0" \
+	"initrd_high=0xffffffff\0" \
+	"fdt_file=imx7d-pico.dtb\0" \
+	"fdt_addr=0x83000000\0" \
+	"ip_dyn=yes\0" \
+	UPDATEHUB_ENV
 
 #define CONFIG_SYS_MEMTEST_START	0x80000000
 #define CONFIG_SYS_MEMTEST_END		(CONFIG_SYS_MEMTEST_START + 0x20000000)
-- 
2.7.4

